networks/p2p/server.go

```diff
@@ -499,7 +498,18 @@ func (srv *MultiChannelServer) SetupConn(fd net.Conn, flags connFlag, dialDest *
 		return errors.New("shutdown")
 	}
 
-	c := &conn{fd: fd, transport: srv.newTransport(fd), flags: flags, conntype: common.ConnTypeUndefined, cont: make(chan error), portOrder: PortOrderUndefined}
+	// retrieve pubkey. if err occurs, dialPubkey is automatically set as nil
+	var dialPubkey *ecdsa.PublicKey
+	if dialDest != nil {
+		dialPubkey, _ = dialDest.ID.Pubkey()
+	}
+
+	c := &conn{fd: fd, flags: flags, conntype: common.ConnTypeUndefined, cont: make(chan error), portOrder: PortOrderUndefined}
```

Comment:
If we move this line above line 501, we can simplify the if statement to check for `dialDest` is `nil`.

Reply:
Yes, refactored at https://github.com/klaytn/klaytn/pull/1954/commits/8e6351fe338cdd3575aacba14bf3120d121ee966.